Reemit.Decompiler.Clr/Metadata/Streams/MetadataTablesStream.cs

```diff
@@ -62,8 +67,16 @@ public MetadataTablesStream(BinaryReader reader)
         MethodDef = ReadTableIfExists<MethodDefRow>();
     }
 
-    private MetadataTable<T>? ReadTableIfExists<T>() where T : IMetadataTableRow<T> =>
-        !_rowsCounts.TryGetValue(T.TableName, out var count)
-            ? null
-            : new MetadataTable<T>(count, _metadataTableDataReader);
+    private MetadataTable<T>? ReadTableIfExists<T>() where T : IMetadataTableRow<T>
+    {
+        if (!_rows.TryGetValue(T.TableName, out var list))
+        {
+            return null;
+        }
+
+        var table = new MetadataTable<T>((uint)list.Capacity, _metadataTableDataReader);
```

Comment:
This is intentional. At this point the list is empty and `Count == 0`. As you can see [here](https://github.com/v0idzz/Reemit/pull/17/files#diff-8bc097c895cb3a74f1aa7894b591a8d06530233698fcf39504e1b81576cde643L44), each list is initialized to have capacity equal to number of records in its corresponding table. I use the `List<T>.Capacity` property to avoid introducing another field to store number of records in tables.